transformer_engine/pytorch/csrc/comm_gemm_overlap.h

```diff
@@ -192,12 +187,17 @@ struct UbufCommOverlap : torch::CustomClassHolder, UbufBase {
       printf("!!! [UB] Register UBuf %d\n", _ub_reg);
     }
 
+    // if priorities are zero, use max priority for comm and min priority for GEMM
+    if (gemm_priority == 0 && comm_priority == 0)
+      std::tie(gemm_priority, comm_priority) = transformer_engine::cuda::stream_priority_range();
+    set_stream_priority(_stream_comm, comm_priority);
     at::cuda::CUDAStream stream_main = at::cuda::getCurrentCUDAStream();
     for (int i = 0; i < std::min(num_max_streams, num_splits); i++) {
-      cudaStream_t stream;
-      cudaStreamCreateWithPriority(&stream, cudaStreamNonBlocking, -1);
+      cudaStream_t stream_tmp;
+      NVTE_CHECK_CUDA(
+          cudaStreamCreateWithPriority(&stream_tmp, cudaStreamNonBlocking, gemm_priority));
       _stream_compute.push_back(
-          at::cuda::getStreamFromExternal(stream, stream_main.device_index()));
+          at::cuda::getStreamFromExternal(stream_tmp, stream_main.device_index()));
```

Comment:
Why do we need to tell pyTorch about our temporary stream?

Reply:
These aren't temporary. We maintain them throughout the lifetime of the Userbuffer object. That said, I don't know why the original code was written to wrap them into `at::cuda::CUDAStream` objects before storing into the vector of compute streams. I suspect that it might be necessary to expose the streams to the PyTorch profiler, but I don't think it makes any performance or functionality difference otherwise whether we do this or not.